Lib/test/test_smtplib.py

```diff
@@ -719,7 +723,7 @@ def _auth_plain(self, arg=None):
                 self.push('535 Splitting response {!r} into user and password'
                           ' failed: {}'.format(logpass, e))
                 return
-            self._authenticated(user, password == sim_auth[1])
+            self._authenticated(user, password == self.sim_auth[1])
```

Comment:
Using this `sim_auth` class attribute and changing it in test code is prone to side-effects, such as different tests affecting each other in a way that depends on the order they are run.

IMO it would be clearer if this were just a global variable.

Regardless, the tests should make sure to reset the global variable back to its default value when they are done, using a a helper method (utilizing `.addCleanup()`) or a context manager.